experimental/dds/tree2/src/feature-libraries/typed-schema/flexList.ts

```diff
@@ -3,39 +3,79 @@
  * Licensed under the MIT License.
  */
 
+/** @remarks Used by `markEager`and `isMarkedEager`. */
+const flexListEager = Symbol("FlexList Eager");
+
+/** Mark the given object as an "eager" item in a `FlexList`, regardless of its type. */
+export function markEager<T>(t: T): T {
+	return isLazy(t)
+		? Object.defineProperty(t, flexListEager, {
+				value: true,
+				configurable: true,
+				enumerable: false,
+				writable: false,
+		  })
+		: t;
+}
+
+/** Check if the given object has been marked "eager" via `markEager`. */
+function isMarkedEager(x: unknown): x is { [flexListEager]: true } {
+	return (x as { [flexListEager]?: boolean })[flexListEager] ?? false;
+}
+
 /**
- * Flexible way to list values.
- * Can be values, functions that return the value (to allow cyclic references to work), or arrays.
- * @remarks
- * Does not work properly if T can be a function.
+ * A flexible way to list values.
+ * Each item in the list can either be a **value** or a **function that returns a value** (the latter allows cyclic references to work).
+ * @privateRemarks
+ * By default, items that are of type `"function"` will be considered "lazy" and all other items will be considered eager.
+ * To force a `"function"` item to be treated as an eager item, call `markEager` before putting it in the list.
+ * This is necessary e.g. when the eager list items are function types and the lazy items are functions that _return_ function types.
+ * `FlexList`s are processed by `normalizeFlexList` and `normalizeFlexListEager`.
  * @alpha
  */
 export type FlexList<Item = unknown> = readonly LazyItem<Item>[];
 
-export function normalizeFlexList<List extends FlexList>(t: List): FlexListToLazyArray<List> {
+/**
+ * Given a `FlexList` of eager and lazy items, return an equivalent list where all items are lazy.
+ */
+export function normalizeFlexListLazy<List extends FlexList>(t: List): FlexListToLazyArray<List> {
 	return t.map((value: LazyItem) => {
-		if (typeof value === "function") {
+		if (isMarkedEager(value)) {
+			return () => value;
+		}
+		if (isLazy(value)) {
```

Comment:
shouldn't isLazy check for eager marks so we don't need both of these cases?

Reply:
Yes, good idea.